dev_guide/builds/triggering_builds.adoc

```diff
@@ -47,27 +47,57 @@ not match, no build is triggered.
 `oc new-app` and `oc new-build` will create GitHub and Generic webhook triggers automatically, but any other needed webhook triggers must be added manually (see xref:setting-triggers[Setting Triggers]).
 ====
 
+For all webhooks, you must define a `Secret` with a key named `WebHookSecretKey` and the
+value being the value to be supplied when invoking the webhook.  The webhook definition
+must then reference the secret.  The secret ensures the uniqueness of the URL, preventing
+others from triggering the build.  The value of the key will be compared to the secret provided 
+during the webhook invocation.
+
+For example here is a GitHub webhook with a reference
+to a secret named `mysecret`:
+
+[source,yaml]
+----
+type: "GitHub"
+github:
+  secretRefence: "mysecret"
+----
+
+The secret is then defined as follows.  Note that the value of the secret is base64 encoded as
+is required for any `data` field of a `Secret` object.
+
+[source,yaml]
+----
+- kind: Secret
+  apiVersion: v1
+  metadata:
+    name: mysecret
+    creationTimestamp: 
+  data:
+    WebHookSecretKey: c2VjcmV0dmFsdWUx
```

Comment:
Where does this data come from?

Reply:
it's the base64 encoded value of the webhook secret.  just like any k8s secret key/value.

```
$ base64 -d
c2VjcmV0dmFsdWUx
^D
secretvalue1
```

Reply:
OK, thanks for reply : )